File: Factorial-20/Factorial20.pas
program Factorial20;

var
  i      : longint;
  value  : integer;
  result : longint;

function factorial( value : integer ) : longint;
var
  result    : longint;
  iteration : integer;
begin
  result := 1;
  for iteration := 1 to value do
  begin
    result := result * iteration;
  end;

  factorial := result;
end;

begin
  for i := 0 to 50000 do
  begin
    for value := 0 to 20 do
    begin
      result := factorial( value );
      WriteLn( result );
    end;
  end;
end.

File: Factorial-20/factorial-20.pl
#!/bin/env perl

use strict;
use warnings;

sub factorial {
  my $value  = shift;
  my $result = 1;

  for ( my $iteration = 1; $iteration <= $value; $iteration++ ) {
    $result = $result * $iteration;
  }

  return $result;
}

if ( ! caller ) {
  for ( my $i = 0; $i <= 50000; $i++ ) {
    for ( my $value = 0; $value <= 20; $value++ ) {
      my $result = factorial( $value );
      print "$result\n"
    }
  }
}
